artifact.go

```diff
@@ -203,67 +213,423 @@ func (m artifactModule) getBuildTestsResultsHandler(c *gin.Context) {
+		ginutil.WriteDBWriteError(c, err, fmt.Sprintf(
+			"Failed saving test result details for build with ID %d in database.",
+			buildID))
+	}
+
+	c.JSON(http.StatusOK, summaryList)
+}
+
+// getBuildAllTestResultDetailsHandler godoc
+// @summary Get all test result details for specified build
+// @tags artifact
+// @param buildid path int true "Build ID"
+// @success 200 {object} []TestResultDetail
+// @failure 400 {object} problem.Response "Bad request"
+// @failure 502 {object} problem.Response "Database is unreachable"
+// @router /build/{buildid}/test-result-details [get]
+func (m artifactModule) getBuildAllTestResultDetailsHandler(c *gin.Context) {
+	buildID, ok := ginutil.ParseParamUint(c, buildIDParamName)
+	if !ok {
+		return
+	}
+
+	details := []TestResultDetail{}
+	err := m.Database.
+		Where(&TestResultDetail{BuildID: uint(buildID)}).
+		Find(&details).
+		Error
 
-	form, err := c.MultipartForm()
 	if err != nil {
-		ginutil.WriteMultipartFormReadError(c, err, fmt.Sprintf(
-			"Failed reading multipart-form content from request body when uploading new artifact for build with ID %d.",
+		ginutil.WriteDBReadError(c, err, fmt.Sprintf(
+			"Failed fetching test result details for build with ID %d from database.",
 			buildID))
 		return
 	}
 
+	c.JSON(http.StatusOK, details)
+}
+
+// getBuildTestResultDetailsHandler godoc
+// @summary Get test result details for specified test
+// @tags artifact
+// @param buildid path int true "Build ID"
+// @param artifactId path int true "Artifact ID"
+// @success 200 {object} []TestResultDetail
+// @router /build/{buildid}/test-result-details/{artifactId} [get]
+func (m artifactModule) getBuildTestResultDetailsHandler(c *gin.Context) {
+	buildID, ok := ginutil.ParseParamUint(c, buildIDParamName)
+	if !ok {
+		return
+	}
+
+	artifactID, ok := ginutil.ParseParamUint(c, artifactIDParamName)
+	if !ok {
+		return
+	}
+
+	details := []TestResultDetail{}
+	err := m.Database.
+		Where(&TestResultDetail{BuildID: uint(buildID), ArtifactID: artifactID}).
+		Find(&details).
+		Error
+
+	if err != nil {
+		ginutil.WriteDBReadError(c, err, fmt.Sprintf(
+			"Failed fetching test result details from test with ID %d for build with ID %d from database.",
+			artifactID, buildID))
+		return
+	}
+
+	c.JSON(http.StatusOK, details)
+}
+
+// getBuildTestResultsSummaryHandler godoc
+// @summary Get test result summary of all tests for specified build
+// @tags artifact
+// @param buildid path int true "Build ID"
+// @success 200 {object} TestResultListSummary
+// @failure 400 {object} problem.Response "Bad Request"
+// @failure 502 {object} problem.Response "Bad Gateway"
+// @router /build/{buildid}/test-results-summary [get]
+func (m artifactModule) getBuildTestResultsSummaryHandler(c *gin.Context) {
+	buildID, ok := ginutil.ParseParamUint(c, buildIDParamName)
+	if !ok {
+		return
+	}
+
+	summaries := []TestResultSummary{}
+	err := m.Database.
+		Preload("Artifact").
+		Where(&TestResultSummary{BuildID: uint(buildID)}).
```

Comment:
Is this cast really needed?

```suggestion
		Where(&TestResultSummary{BuildID: buildID}).
```

_hint: Running `npm run lint-go` will probably reveal these unneeded casts, as well as other minor scripting errors_